### PR TITLE
Update documentation for solarlog

### PR DESCRIPTION
**Pull request by dontinelli**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/123205
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Clarified security requirements for activating the open JSON interface, emphasizing the necessity of setting a user password.
	- Added a new section detailing conditions for accessing additional data and associated security risks.
	- Provided guidance on enabling options to receive extra data from the solarlog device, including information about new available sensors.

### CODE REVIEW

File: source/_integrations/solarlog.markdown [hunk 1]
```diff
@@ -75,6 +75,29 @@ The following sensors are available in the library:
 | power_available       | W      | Available power (equals to power_ac - consumption_ac) | 
 | usage                 | %      | Usage (equals to consumption_ac / power_ac) |
 
+## Additional data
+
+{% important %}
+The additional data is only accessible, if password protection for the user is deactivated. Obviously this is a security issue and should only be done in specific circumstanses. In any event, you do this at your own risk.
```
Comment (c0ffeeca7) [inline source/_integrations/solarlog.markdown]:
```suggestion
The additional data is only accessible if the user's password protection is deactivated. Obviously, deactivating password protection is a security risk and should only be done in specific circumstances. In any event, you do this at your own risk.
```

File: source/_integrations/solarlog.markdown [hunk 2]
```diff
@@ -75,6 +75,29 @@ The following sensors are available in the library:
...
+{% endimportant %}
+
+Users may choose to get additional data from the solarlog device. To enable, check the box for extended data in the system options of the integration.
```
Comment (c0ffeeca7) [inline source/_integrations/solarlog.markdown]:
```suggestion
You can get additional data from the Solar-Log device. To enable this, select the checkbox for extended data in the integration's system options.
```

File: source/_integrations/solarlog.markdown [hunk 3]
```diff
@@ -75,6 +75,29 @@ The following sensors are available in the library:
...
+
+The following additional sensor becomes available:
+
+| name                  | Unit   | Description   |
+|-----------------------|--------|:-------------------------------------------|
+| self_consumption_year | kWh    | Annual self-consumed solar power.          |
+
+In addition, you can choose to get additional information on devices that are connected to the Solar-Log device. For this, check the box of the respective device in the configuration dialog of the integration.
```
Comment (c0ffeeca7) [inline source/_integrations/solarlog.markdown]:
```suggestion
In addition, you can choose to get additional information on devices that are connected to the Solar-Log device. For this, select the checkbox of the respective device in the configuration dialog of the integration.
```

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @dontinelli :+1:

**Review by dontinelli — COMMENTED** [review]
Thank you for the comments and adjustments!

File: source/_integrations/solarlog.markdown [hunk 4]
```diff
@@ -75,6 +75,27 @@ The following sensors are available in the library:
...
+
+{% important %}
+The additional data is only accessible if the user's password protection is deactivated. Obviously, deactivating password protection is a security risk and should only be done in specific circumstances. In any event, you do this at your own risk.
+{% endimportant %}
+
+You can get additional data from the Solar-Log device. To enable this, select the checkbox for extended data in the integration's system options.
+
+The following additional sensor becomes available:
+
+| name                  | Unit   | Description   |
```
Comment (c0ffeeca7) [inline source/_integrations/solarlog.markdown]:
```suggestion
| Name                  | Unit   | Description   |
```

File: source/_integrations/solarlog.markdown [hunk 5]
```diff
@@ -75,6 +75,27 @@ The following sensors are available in the library:
...
+|-----------------------|--------|:-------------------------------------------|
+| self_consumption_year | kWh    | Annual self-consumed solar power.          |
+
+In addition, additional information on devices that are connected to the Solar-Log device become available. The following additional sensors are available (all values are per inverter/device):
+
+| name                  | Unit   | Description   |
```
Comment (c0ffeeca7) [inline source/_integrations/solarlog.markdown]:
```suggestion
| Name                  | Unit   | Description   |
```

File: source/_integrations/solarlog.markdown [hunk 6]
```diff
@@ -75,6 +75,27 @@ The following sensors are available in the library:
...
+
+You can get additional data from the Solar-Log device. To enable this, select the checkbox for extended data in the integration's system options.
+
+The following additional sensor becomes available:
+
+| name                  | Unit   | Description   |
+|-----------------------|--------|:-------------------------------------------|
+| self_consumption_year | kWh    | Annual self-consumed solar power.          |
+
+In addition, additional information on devices that are connected to the Solar-Log device become available. The following additional sensors are available (all values are per inverter/device):
```
Comment (c0ffeeca7) [inline source/_integrations/solarlog.markdown]:
```suggestion
In addition, information from devices connected to the Solar-Log device becomes available. The following additional sensors become available (all values are per inverter/device):
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @dontinelli :+1:

**Pull request merged**